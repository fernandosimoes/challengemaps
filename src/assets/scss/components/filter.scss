%defaultinput {
  border-radius: 5px;
  color: $gray;
  font-size: 20px;
  height: 35px;
  margin-bottom: 15px;
  padding: 0 15px;
  text-align: right;
  width: 320px;

  &::placeholder {
    color: $lightgray;
  }
}

.minimumvalue {
  label {
    font-family: $latobold
  }
}

.minimumvalueinput {
  @extend %defaultinput;
}

.searchinput {
  @extend %defaultinput;
  align-self: center;
  text-align: left;
  width: 100%;
}

.filters {
  justify-content: space-between;
  width: 100%;

  .search {
    align-items: center;
    display: flex;
    width: 100%;
  }
}

.searchinput {
  background: no-repeat 98% center;
  margin-top: 28px;
}

.faturamentominimo {
  align-items: center;
  display: flex;
  font-size: 18px;
  justify-content: space-between;

  label {
    margin-bottom: 15px;
  }

  button {
    margin-top: 10px;
  }

  span {
    color: $red;
    font-size: 12px;
    margin: -15px 0 15px;
  }
}