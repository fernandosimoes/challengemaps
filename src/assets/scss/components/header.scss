.header {
  background: $blue;
  font-size: 35px;
  margin-bottom: 25px;
  width: 100%;

  h1 {
    color: $white;
    font-family: $lato;
    padding: 25px;
  }
}