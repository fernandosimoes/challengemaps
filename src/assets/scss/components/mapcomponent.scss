.mapcontent {
    width: 40%;
}

.minimal--value {
    border-radius: 5px;
    color: $gray;
    height: 35px;
    margin-bottom: 15px;
    text-align: right;
    width: 320px;
    padding: 0 15px;
    font-size: 20px;
    &::placeholder {
        color: $lightgray;
    }
}

.faturamentominimo {
    display: flex;
    flex-direction: column;
    font-size: 25px;

    label {
        margin-bottom: 15px;
    }

    span {
        margin: -15px 0 15px;
        font-size: 12px;
        color: $red;
    }
}