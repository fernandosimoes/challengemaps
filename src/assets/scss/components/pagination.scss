.pagination {

  button {
    background: none;
    border: none;
    cursor: pointer;

    &.selected {
      background-color: $lightgray;

    }
  }
}
