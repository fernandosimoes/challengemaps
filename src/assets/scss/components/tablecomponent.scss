.tablecontent {
    width: 100%;
}

.table {
    border-bottom: 0;
    border-radius: 5px;
    border: 1px solid $gray;

    .line {
        border-bottom: 1px solid $gray;
        display: flex;
        justify-content: space-between;
        width: 100%;
        span {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
        }

        a {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
            text-decoration: none;
            color: $gray;
            font-weight: bold;
        }

        .low {
            color: $red
        }
    }
}