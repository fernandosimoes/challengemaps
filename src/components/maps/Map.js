import React, {Component} from "react";
import { compose, withProps } from "recompose";
import {
    withScriptjs,
    withGoogleMap,
    GoogleMap,
    Marker,
} from "react-google-maps";
import { InfoBox } from "react-google-maps/lib/components/addons/InfoBox";
//?key=AIzaSyBkNaAGLEVq0YLQMi-PYEMabFeREadYe1Q&


const mappositions = [
    { "name": "Alameda Santos", "city": "São Paulo", "state": "SP", "latitude": -23.568767, "longitude": -46.649907, "revenue": 29854.12 },
    { "name": "Av Paulista", "city": "São Paulo", "state": "SP", "latitude": -23.565972, "longitude": -46.650859, "revenue": 31257.23 },
    { "name": "Av Paulista Novo", "city": "São Paulo", "state": "SP", "latitude": -23.563703, "longitude": -46.653542, "revenue": 19874.25 },
    { "name": "Bandeirantes", "city": "São Paulo", "state": "SP", "latitude": -23.616359, "longitude": -46.664749, "revenue": 21587.33 },
    { "name": "Barão do Triunfo", "city": "São Paulo", "state": "SP", "latitude": -23.624203, "longitude": -46.683369, "revenue": 12569.95 },
    { "name": "Bela Vista", "city": "São Paulo", "state": "SP", "latitude": -23.565972, "longitude": -46.650859, "revenue": 22365.54 },
]

class Map extends Component {
    render() {
        withGoogleMap()
        withScriptjs()
        return (
            <div>
                <GoogleMap defaultZoom={10} defaultCenter={{ lat: -23.56, lng: -46.644 }}>
                    {this.props.isMarkerShown && (
                        mappositions.map((map, key) => {
                            return (<Marker key={key} position={{ lat: map.latitude, lng: map.longitude }}>
                                <InfoBox
                                    options={{ closeBoxURL: ``, enableEventPropagation: true }}
                                >
                                    <div style={{ backgroundColor: `yellow`, opacity: 0.75, padding: `12px` }}>
                                        <div style={{ fontSize: `16px`, fontColor: `#08233B` }}>
                                            Hello, Kaohsiung!
                            </div>
                                    </div>
                                </InfoBox>
                            </Marker>)
                        })
                    )}
                </GoogleMap>

            </div>
        );
    }
}

export default Map;

{/* <Marker
    position={{ lat: 22.6273, lng: 120.3014 }}
    onClick={props.onToggleOpen}
>
    {props.isOpen && <InfoBox
        onCloseClick={props.onToggleOpen}
        options={{ closeBoxURL: ``, enableEventPropagation: true }}
    >
        <div style={{ backgroundColor: `yellow`, opacity: 0.75, padding: `12px` }}>
            <div style={{ fontSize: `16px`, fontColor: `#08233B` }}>
                Hello, Kaohsiung!
          </div>
        </div>
    </InfoBox>}
</Marker> */}
// class Map extends Component {
//     render() {
//         return (
//             <div>
                
//             </div>
//         );
//     }
// }
