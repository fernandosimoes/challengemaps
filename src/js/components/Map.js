import React, { Component } from 'react';
import { Map, InfoWindow, Marker, GoogleApiWrapper } from 'google-maps-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';


import redmarker from  "../../assets/images/marker-red.png";
import bluemarker from  "../../assets/images/marker-blue.png";

const style = {
    position: 'relative'
}

class CustomMap extends Component {
    constructor(props) {
        super(props);
        this.state = {
            errorvalue: false,
            showingInfoWindow: false,
            activeMarker: {},
            selectedPlace: {},
            allStore: true,
        };
        this.includeMarkers = this.includeMarkers.bind(this);
        this.onMarkerClick = this.onMarkerClick.bind(this);
        this.onMapClicked = this.onMapClicked.bind(this);
        this.changeMinimalValue = this.changeMinimalValue.bind(this);
        this.justShowStorespage = this.justShowStorespage.bind(this);
    }

    justShowStorespage() {
        if(this.state.allStore) {
            this.setState({
                allStore: false
            })
        } else {
            this.setState({
                allStore: true
            })
        }
    }
    changeMinimalValue(e) {
        if (Number(e.target.value) !== "NaN") {
            this.setState({
                minimalvalue: e.target.value,
                errorvalue: false
            })
        } else {
            this.setState({
                errorvalue: true
            })

        }
    }

    onMarkerClick (props, marker, e){
        this.setState({
            selectedPlace: props,
            activeMarker: marker,
            showingInfoWindow: true
        });
    }
    onMapClicked(props){
        if (this.state.showingInfoWindow) {
            this.setState({
                showingInfoWindow: false,
                activeMarker: null
            })
        }
    };

    includeMarkers(lojas) {
        if (!lojas.length) {
            return false;
        }
        return lojas.map((store, key)=> {
            if (store.revenue < this.props.faturamentoMinimo) {
                return (
                <Marker
                    key={key}
                    onClick={this.onMarkerClick}
                    name={store.name}
                    position={{ lat: store.latitude, lng: store.longitude }}
                    icon={{
                        url: redmarker,
                        // anchor: new google.maps.Point(32, 32),
                        // scaledSize: new google.maps.Size(64, 64)
                    }}
                    />
                )
            } else {
                return <Marker
                    key={key}
                    onClick={this.onMarkerClick}
                    name={store.name}
                    position={{ lat: store.latitude, lng: store.longitude }}
                    icon={{
                        url: bluemarker,
                        // anchor: new google.maps.Point(32, 32),
                        // scaledSize: new google.maps.Size(64, 64)
                    }}
                    />

            }
        })
    }

    shouldComponentUpdate (nextProps, nextState){
        // if (this.props.pagesonmap != nextProps.pagesonmap && nextProps.filter) {
            // }
            // return false;
                return true;
    };

    render() {
        const errorclass = this.state.errorvalue === false ? 'hide' : 'show';
        return (
            <div className="mapcontent">
                <Map
                    google={this.props.google}
                    zoom={12}
                    containerStyle={style}
                    className=""
                    initialCenter={{
                        lat: -23.533773,
                        lng: -46.625290
                    }}
                    >
                    {this.props.pagesonmap && this.includeMarkers(this.props.splitedPages[this.props.currentPage])}

                    {!this.props.pagesonmap && this.includeMarkers(this.props.lojas)}
                    <InfoWindow
                        marker={this.state.activeMarker}
                        visible={this.state.showingInfoWindow}>
                        <div>
                            <h1>{this.state.selectedPlace.name}</h1>
                        </div>
                    </InfoWindow>
                </Map>
            </div>
        );
    }
    componentDidMount() {
        // this.includeMarkers();
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        lojas: state.lojas,
        currentPage: state.currentPage,
        splitedPages: state.splitedPages,
        faturamentoMinimo: state.minimumvalue,
        filter: state.filter,
        pagesonmap: state.pagesonmap
    }
}


const mapDispatchToProps = (dispatch, ownProps) => {
    return { }
}



const LoadingContainer = (props) => (
    <div className="teste">Fancy loading container!</div>
)
const googleapi = GoogleApiWrapper({ apiKey: '', LoadingContainer: LoadingContainer })(CustomMap)
const MapComponent = connect(mapStateToProps, mapDispatchToProps)(googleapi) // used in a real render of this SPA. the other option CustomMap stand alone is used in the test with jest

export {MapComponent, CustomMap}

