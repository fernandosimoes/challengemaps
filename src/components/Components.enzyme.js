// Link.react.test.js
import React from 'react';
import Header from './Header';
import { MapContainer } from './Map';

import { configure, shallow, mount, render } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Enzyme Map', () => {
    // Render a checkbox with label in the document
    const map = shallow(<MapContainer />);
    // expect(header.hasClass('header')).toBe(true);
    // expect(wrapper.find('.my-button').hasClass('disabled')).to.equal(true);
    // expect(checkbox.text()).toEqual('Off');

    // checkbox.find('input').simulate('change');

    // expect(checkbox.text()).toEqual('On');
});