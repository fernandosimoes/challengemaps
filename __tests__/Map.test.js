import React from 'react';
import { CustomMap } from '../src/js/components/Map';
import renderer from 'react-test-renderer';
import {allstores} from './fakedata';
test('Map component', () => {

    let component = renderer.create(
        <CustomMap pagesonmap={false} lojas={allstores.stores} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
