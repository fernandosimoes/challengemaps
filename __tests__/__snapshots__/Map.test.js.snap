// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component 1`] = `
<div
  className="mapcontent"
>
  <div
    className=""
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "display": "inherit",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      Loading map...
    </div>
  </div>
</div>
`;
